objgraph.py

```diff
@@ -65,6 +65,15 @@
     iteritems = dict.items
 
 
+def _isinstance(object, classinfo):
+    """Use an implementation of isinstance that doesn't depend on ``__class__``,
+    which is delegated to the mock spec by the mocking library."""
+    if sys.version_info[0] == 2:
+        return issubclass(type(object), classinfo)
+    else:
+        return isinstance(object, classinfo)
```

Comment:
I'm curious why no special handling is required on Python 3.

I'm also tempted to optimize prematurely, i.e.

```python
if sys.version_info[0] >= 3:
    _isinstance = isinstance
else:
    def _isinstance(obj, classinfo):
        """...

        ...
        """
        return issubclass(type(obj), classinfo)
```